addons/website_event_sale/tests/test_website_event_sale.py

```diff
@@ -5,32 +5,49 @@
 
 class TestWebsiteEventSale(HttpCaseWithUserPortal, TestWebsiteEventSaleCommon):
 
-    def test_website_event_sale(self):
-        """ Test saleorder is created for tickets having price only """
-        self.authenticate('portal', 'portal')
+    def test_website_event_sale_free_tickets(self):
+        """ Test saleorder is not created for tickets free tickets """
+        self.authenticate(None, None)
         free_ticket = self.env['event.event.ticket'].create({
             'event_id': self.event.id,
             'name': 'Free',
             'product_id': self.product_event.id,
             'price': 0,
         })
         event_questions = self.event.question_ids
+        event_registration_count = len(self.event.registration_ids)
         name_question = event_questions.filtered(lambda q: q.question_type == 'name')
         email_question = event_questions.filtered(lambda q: q.question_type == 'email')
         phone_question = event_questions.filtered(lambda q: q.question_type == 'phone')
+        existing_so = self.env['sale.order'].search([
+            ('order_line.event_ticket_id', '=', free_ticket.id)
+        ])
         self.url_open(f'/event/{self.event.id}/registration/confirm', data={
             f'1-name-{name_question.id}': 'Bob',
             f'1-email-{email_question.id}': 'bob@test.lan',
             f'1-phone-{phone_question.id}': '8989898989',
             '1-event_ticket_id': free_ticket.id,
             'csrf_token': http.Request.csrf_token(self),
         })
-        self.assertFalse(self.env['sale.order'].search([
-            ('partner_id', '=', self.partner_portal.id),
+        self.assertEqual(self.env['sale.order'].search([
```

Comment:
Actually it should still be assertFalse, isn't it ? Or, better, check DB state before and after the test, like
```
    init_so = self.env['sale.order'].search([])
    # do stuff
    ....
    post_so = self.env['sale.order'].search([])
    self.assertEqual(post_so, init_so, "No new SO should have been created during the flow")
```